File: javascript/int_stack.js
const Stack = require('./stack');

class IntStack extends Stack {
  constructor() {
    super();
    this.isInt = (str) => {
      const regex = new RegExp(/^\d+$/gi);
      return regex.test(str);
    };
  }

  // Overriding method from SuperClass in the SubClass
  push(item) {
    if (this.isInt(item.toString())) {
      this.items.push(item);
    } else {
      console.error(
        `Invalid item: ${item} entered. This stack accepts integers only`
      );
    }
  }
}

File: ruby/int_stack.rb
require './stack'

class IntStack < Stack
  def push(item)
    if is_int?(item)
      @items.push(item)
    else
      puts "Invalid item: #{item} entered. This stack accepts integers only"
    end
  end

  private

  def is_int?(item)
    item.to_s.match(/^\d+$/i)
  end
end
